src/invidious/yt_backend/connection_pool.cr

```diff
@@ -45,24 +42,22 @@ struct YoutubeConnectionPool
 
   private def build_pool
     DB::Pool(HTTP::Client).new(initial_pool_size: 0, max_pool_size: capacity, max_idle_pool_size: capacity, checkout_timeout: timeout) do
-      conn = HTTP::Client.new(url)
-      conn.family = CONFIG.force_resolve
+      conn = make_client(url, force_resolve = true)
```

Comment:
It should be that, btw:
```suggestion
      conn = make_client(url, force_resolve: true)
```

I don't understand why the `=` isn't flagged by the compiler as invalid syntax.

Reply:
So, from a quick discussion in the Crystal matrix room, doing
```crystal
make_client(url, foo = true)
```
is the same as doing
```crystal
foo = true
make_client(url, foo)
```
i.e it assigns the value to a local variable and then passes that variable as a positional function parameter. In this case, this would be end up being assigned to `region`.

Reply:
Oops thanks!

I guess that also explains why this wasn't caught my the CI https://github.com/iv-org/invidious/pull/4709#discussion_r1612491743

Reply:
Yep! I've opened an issue at the Crystal bug tracker, because it seems like something that shouldn't be allowed by design: https://github.com/crystal-lang/crystal/issues/14722